### PR TITLE
ARTEMIS-2339 Compatibility around prefixing

### PR DESCRIPTION
**Pull request by clebertsuconic**
There are a few issues with prefixing and compatibility.
This is basically an issue when integrated with Wildfly or any other case
where prefix is activated
and playing with older versions.

### CODE REVIEW

File: artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java [hunk 1]
```diff
@@ -51,4 +52,10 @@ public ActiveMQTextCompabileMessage(ClientMessage message, ClientSession session
    public ActiveMQTextCompabileMessage(TextMessage foreign, ClientSession session) throws JMSException {
       super(foreign, session);
    }
+
+
+   @Override
+   protected SimpleString checkPrefix(SimpleString address) {
+      return ActiveMQCompatibleMessage.checkPrefix1X(address);
```
Comment (michaelandrepearce) [inline artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: Should this be on all compatible1x messages

Comment (clebertsuconic) [reply artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: It is.. that's what ActiveMQTextCompatibleMessage is

Comment (michaelandrepearce) [inline artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: Should this be on all compatible1x messages

Comment (clebertsuconic) [reply artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: ah.. this one was missing. .it was a typo

Comment (clebertsuconic) [reply artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: this one was missing.. consider it a typo.. it's the actual issue

Comment (graben) [reply artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: Should the type of the class name itself be fixed, too? (ActiveMQText**Compabile**Message.java)

Comment (michaelandrepearce) [reply artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: E.g. why is ActiveMQTextCompabile special vs others like BytesMessage or ObjectMessage

File: artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java [hunk 2]
```diff
@@ -51,4 +52,10 @@ public ActiveMQTextCompabileMessage(ClientMessage message, ClientSession session
    public ActiveMQTextCompabileMessage(TextMessage foreign, ClientSession session) throws JMSException {
```
Comment (michaelandrepearce) [inline artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: Not part of this change i know. But spotted spelling mistake. Compabile should be Compatible

File: artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java (hunk 1 above)
Comment (clebertsuconic) [reply artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: @graben ouch.. yeah.. I will fix it

Comment (clebertsuconic) [reply artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: @michaelandrepearce I don't understand your question... all the other classes had checkPrefix added, only TextMessage was missing.

Comment (michaelandrepearce) [reply artemis-jms-client/src/main/java/org/apache/activemq/artemis/jms/client/compatible1X/ActiveMQTextCompabileMessage.java]: That was my question. You answered it :)

**Comment by clebertsuconic** [comment]
this is ready to merge now, as long as we get a Green result on PR-check. 

Had a few failures along the review here, that were basically test issues. All fixed now.
